### PR TITLE
Tock 2.0: update sound pressure driver

### PR DESCRIPTION
**Pull request by alexandruradovici**
### Pull Request Overview

This pull request update the sound pressure driver to the new 2.0 driver API.

Using the new API requires increasing the kernel stack size (it faults with stack overflow otherwise).

### Testing Strategy

This pull request was tested using a MicroBit v2.

### TODO or Help Wanted

N/A

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by hudson-ayers — APPROVED** [review]
very straightforward

**Review by phil-levis — APPROVED** [review]
Looks good. But why did the stack size increase?

File: boards/microbit_v2/src/main.rs [hunk 1]
```diff
@@ -71,7 +71,7 @@ static mut CHIP: Option<&'static nrf52833::chip::NRF52<Nrf52833DefaultPeripheral
 /// Dummy buffer that causes the linker to reserve enough space for the stack.
 #[no_mangle]
 #[link_section = ".stack_buffer"]
-pub static mut STACK_MEMORY: [u8; 0x1500] = [0; 0x1500];
```
Comment (phil-levis) [inline boards/microbit_v2/src/main.rs:74]: Why did this need to increase?

Comment (alexandruradovici) [reply boards/microbit_v2/src/main.rs:74]: It faults with Kernel stack overflow otherweise

**Pull request merged**